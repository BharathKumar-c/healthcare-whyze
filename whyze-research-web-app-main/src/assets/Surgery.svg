<svg width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect width="40" height="40" rx="20" fill="#FFF7E0"/>
<path d="M22.1373 19.8156H20.6373V18.3156C20.6373 17.9406 20.3373 17.6406 19.9623 17.6406C19.5873 17.6406 19.2873 17.9406 19.2873 18.3156V19.8156H17.7873C17.4123 19.8156 17.1123 20.1156 17.1123 20.4906C17.1123 20.8656 17.4123 21.1656 17.7873 21.1656H19.2873V22.6656C19.2873 23.0406 19.5873 23.3406 19.9623 23.3406C20.3373 23.3406 20.6373 23.0406 20.6373 22.6656V21.1656H22.1373C22.5123 21.1656 22.8123 20.8656 22.8123 20.4906C22.8123 20.1156 22.5123 19.8156 22.1373 19.8156Z" fill="#FABB00"/>
<path d="M31.5498 19.2893C31.5498 17.6393 30.1998 16.2893 28.5498 16.2893H27.3873L27.1248 15.2768C26.9748 14.6768 26.5998 14.1893 26.0748 13.8893C25.5123 13.5893 24.9123 13.5143 24.2748 13.7393C20.8248 14.8268 17.3748 14.1518 15.6123 13.6268C15.0123 13.4768 14.4123 13.5143 13.8873 13.8518C13.3623 14.1518 12.9873 14.6393 12.8373 15.2393L12.5748 16.2893H11.4873C9.8373 16.2893 8.4873 17.6393 8.4873 19.2893C8.4873 20.7893 9.6123 22.0643 11.0373 22.2518L10.9623 22.5518C10.6998 23.6768 11.2998 24.8393 12.3873 25.2143C15.0498 26.1518 17.5998 26.4518 19.8873 26.4518C23.1498 26.4518 25.8498 25.7768 27.5748 25.2143C28.6998 24.8393 29.2998 23.6768 29.0373 22.5518L28.9623 22.2893C30.3873 22.1018 31.5498 20.8268 31.5498 19.2893ZM9.7998 19.2893C9.7998 18.3518 10.5498 17.6018 11.4873 17.6018H12.1998L11.3373 21.0143C10.4748 20.9393 9.7998 20.1893 9.7998 19.2893ZM27.1248 23.9768C24.3873 24.8768 18.9123 26.0768 12.7998 23.9768C12.3498 23.8268 12.1248 23.3393 12.2373 22.8893L14.0748 15.5768C14.1498 15.3143 14.2998 15.1268 14.5248 15.0143C14.6748 14.9393 14.8248 14.9018 14.9748 14.9018C15.0498 14.9018 15.1623 14.9018 15.2373 14.9393C17.1498 15.5018 20.8623 16.2143 24.6498 15.0143C24.9123 14.9393 25.1748 14.9393 25.3998 15.0893C25.6248 15.2018 25.7748 15.4268 25.8498 15.6518L27.7248 22.9268C27.8373 23.3393 27.5748 23.8268 27.1248 23.9768ZM28.5873 21.0143L27.6873 17.6018H28.5123C29.4498 17.6018 30.1998 18.3518 30.1998 19.2893C30.2373 20.2268 29.4873 20.9768 28.5873 21.0143Z" fill="#FABB00"/>
</svg>
