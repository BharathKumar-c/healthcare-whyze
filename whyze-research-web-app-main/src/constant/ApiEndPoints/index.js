const FEASIBILITY_API = process.env.REACT_APP_FEASIBILITY_STUDY_BASE_URL;
const AUTH_API = process.env.REACT_APP_AUTH_URL;

export const feasibilityStudyApiUrls = {
  medicalConditionSearchUrl: `${FEASIBILITY_API}/search/indication`,
  medicationConditionSearchUrl: `${FEASIBILITY_API}/search/medication`,
  caseStudyUrl: `${FEASIBILITY_API}/case`,
  locationUrl: `${FEASIBILITY_API}/case/location`,
  initialPopulation: `${FEASIBILITY_API}/case/get-count`,
  allergiesListSearchUrl: `${FEASIBILITY_API}/search/allergies`,
  reactionListSearchUrl: `${FEASIBILITY_API}/search/reaction`,
  sitesDataUrl: `${FEASIBILITY_API}/case/site`,
  sitesWithPatientsCountUrl: `${FEASIBILITY_API}/case/site-patient-count`,
};

export const authEndPoints = {
  login: `${AUTH_API}/login`,
  register: `${AUTH_API}/register`,
  refreshToken: `${AUTH_API}/refreshToken`,
  forgotPassword: `${AUTH_API}/sendResetPasswordLink`,
  resetPassword: `${AUTH_API}/resetPassword`,
};

export const dashboardApiUrls = {
  projectUrl: `${FEASIBILITY_API}/project`,
  favouriteUrl: `${FEASIBILITY_API}/project/favourite`,
};

export const profileApiUrls = {
  image: `${AUTH_API}/image`,
  user: `${AUTH_API}/user`,
  profilePicture: `${AUTH_API}/profilePicture`,
};
