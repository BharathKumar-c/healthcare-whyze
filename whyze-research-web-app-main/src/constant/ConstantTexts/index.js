import React from 'react';

// dashboard welcome Screen const text

export const welComeComponentContent =
  'See how Whyze is preforming for researchers like you.';
export const createFirstStudy = 'Create your first study';
export const createFirstStudyContent =
  'Define your protocol or create your first feasibility study';
export const patientstitleLable = 'Patients';
export const totalPatientsLable = 'Total patients';
export const countriesLable = 'Countries';
export const sitesLable = 'Sites';
export const sitesRecruitedLable = 'Sites recruited';
export const avgTimeLable = 'Avg. response time';
export const trialsLable = 'Trials';
export const activeTrials = 'Active trials';
export const totalTrials = 'Total trials';
export const screenFailRate = 'Screen fail rate';
export const threeMonths = '3 months';
export const avgTimeToRecruit = 'Avg. time to recruit';
export const feasibilityStudiesInPlanning = 'Feasibility studies in planning';
export const seeAll = 'See all';

//Profile Page const text
export const ResearcherTitle = 'Researcher title -';
export const AuthProfile = 'AUTHENTICATED PROFILE';
export const ContactDetails = 'Contact details';
export const CompanyEmail = 'Company email address';
export const PhoneNumber = 'Phone';

//ContactModal Page Const texts

export const EmailAddress = 'Email address';
export const Phone = 'Phone';
export const okButtonLabel = 'Ok';
export const saveButtonLabel = 'save';
export const cancelButtonLabel = 'cancel';
export const contactDetailsLabel = 'Contact details';

//Addcriteria texts
export const addCriteriaTitle = 'Select a criteria item';

//RenameModal texts
export const renameModalTitle = 'Profile details';
export const fullNameLabel = 'Full name';
export const titleLabel = 'Title';

// Medication constant texts
export const medicationConstants = {
  medicationLabel: 'Medication',
  medicationErrorMessage: 'Please add the Medication',
  selectlabel: 'Select Dosage',
  inputLabel: 'Enter value',
  startedPrefixText: 'Started at least',
};
// Indication constant texts
export const indicationConstants = {
  medicalConditionLabel: 'Medical condition',
  medicalConditionErrorMessage: 'Please add the Medical condition',
};

//Allergies constant texte
export const allergiesConstants = {
  allergiesLabel: 'Allergies criteria',
  allergiesPlaceholder: 'Allergies',
  allergiesErrorText: 'Please add the Allergies',
};
//Symptom constant texts

export const symptomConstats = {
  symptomLabel: 'Symptom criteria',
  symptomPlaceholder: 'Symptom',
};
//Project settingmodal Constants

export const projectSettingConstants = {
  addProjectError: 'Please add the project',
  addProjectLabel: 'Project',
  addClientNameError: 'Please add ClientName',
  rangeLabel: 'Select',
  addClientLabel: 'Client name',
  description: 'Description',
  sharingLabel: 'Sharing',
  shareBtnLabel: 'Share',
  standardsText: 'Standards',
  standardDescription: 'Standardising markers and value outputs',
  measurementText: 'Default measurement',
  clinicalTrialError: 'Please add the clinical trial',
  clinicalTrial: 'Clinical Trial',
  sponsorError: 'Please add the sponsor',
  sponsor: 'Sponsor',
  projectSettings: 'Project settings',
  descriptionError: 'Please add the description',
  standardsError: 'Please select the standards',
  noIndication: 'No Indication',
};
export const shareProjectConstants = {
  addPeoplePlaceHolder: 'Add people and groups',
  accessPeopleText: 'People with access',
};
export const selectedPeoplesConstants = {
  addPeople: 'Add people and groups',
  notifyPeopleLabel: 'Notify people',
  messageLabel: 'Message',
};
// Nicotine constant texts
export const nicotineConstants = {
  nicotineConditionErrorMessage: 'Please add the nicotine condition',
  nicotineLabel: 'Nicotine criteria',
};

// Create Account constant texts
export const createAccountContants = {
  createAccounTitle: (
    <>
      Set up your <br /> Whyze account
    </>
  ),
  setUpWhyzeAccount: 'Set up your Whyze account',
  nameLabel: 'Name',
  emailLabel: 'Company email',
  phoneLabel: 'Your phone number',
  passwordLabel: 'Create password',
  characterLengthLabel: 'Between 8 and 20 characters',
  numericLabel: 'At least one number',
  uppercaseLabel: 'At least one upper-case letter',
  lowercaseLabel: 'At least one lower-case letter',
  createAccountButtonLabel: 'Create account',
  agreeConditionText: 'By clicking the button above you agree to our',
  termsOfUseText: 'Terms of use',
  privacyNoticesText: 'Privacy policies',
  alreadyHaveAccountText: 'Already have an account?',
  loginText: 'Login',
  termsOfUseLink: 'https://www.whyzehealth.com/privacy-notice/',
  privacyNoticesLink: 'https://www.whyzehealth.com/terms',
};

//indication const

export const conditions = {
  started: 'started',
  dosage: 'dosage',
  courseOfTherapy: 'course of therapy',
  diagnosed: 'diagnosed',
  recovered: 'recovered',
  severity: 'severity',
};

//demographics const

export const demographicsConst = {
  age: 'age',
  min: 'min',
  max: 'max',
  gender: 'gender',
};

//sites const

export const siteConst = {
  siteInformationCosnt: {
    patientSegmentHeader: 'Patient segment',
    connectedWithSite: 'Connected with this site',
    additionalCatchmentAreas:
      'Additional Whyze patients in site catchment areas',
    sitePerformance: 'Site performance',
    recruitmentTimeline: 'Recruitment timeline',
    dropOutRate: 'Drop out rate',
    trackRecordOfDoingTrials: 'Track record of doing trials',
    completedInThePastyear: 'Completed in the past year',
    primaryInvestigator: 'Primary Investigator',
    administration: 'Administration',
    screenFailRate: 'Screen fail rate',
    industryAverage: 'Industry average',
    avgResponseTime: 'Avg. response time',
  },
};

//setupAccount const

export const setupAccountConst = {
  dontHaveaccount: 'Don’t have an account yet?',
  setUpAccount: 'Set up account',
};

//forgotpassword const

export const forgotpasswordConst = {
  WHYZE: 'WHYZE',
  forgotPassword: 'Forgot Password',
  resendEmail: 'Enter your email below and we will send you a reset email.',
  pleaseTry: 'Didn’t receive it ? Please try again',
  returnTo: 'return to',
  loginPage: 'login',
};

//weight const  texts

export const weightConst = {
  weightRange: 'Weight Range',
  weightMin: 'min',
  weightMax: 'max',
  weightType: 'number',
  weightmaxErrorText: 'add max weight',
  weightminErrorText: 'add min weight',
  weightFrom: 'From',
  weightTo: 'To',
};

export const SaveAndCloseModalConst = {
  SaveAndCloseModalTitle: 'Name before exiting',
  projectNameInputLabel: 'Project',
  projectNameInputPlaceholder: 'New patient segment',
  clientNameInputLabel: 'Client name',
  descriptionNameInputLabel: 'Description',
  patientSegmentInputLabel: 'Patient segment required',
};

// Proposed Site list constant texts

export const proposedSiteListConstants = {
  title: 'Proposed Site List',
  seeMore: 'See more',
  emptyContent:
    'To create your proposed list, drag & drop or add your selected sites here.',
};

// Surgery constant text
export const surgeryConstants = {
  surgeryInputLabel: 'Surgery criteria',
  surgeryInputErrorMessage: 'Please add the Surgery',
};

export const feasibilityStudy = {
  newFeasibilityStudy: 'new-feasibility-study',
};
