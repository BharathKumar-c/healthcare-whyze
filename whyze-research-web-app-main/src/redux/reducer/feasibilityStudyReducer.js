import { createSlice } from '@reduxjs/toolkit';
import { v4 as uuid } from 'uuid';

export const initialState = {
  segments: [
    {
      name: 'Segment-1',
      isDone: false,
      activeTab: {
        name: 'feasibility',
        index: 0,
      },
      tabs: [
        {
          name: 'feasibility',
          title: 'Feasibility',
          isActive: true,
          position: 'left',
        },
      ],
      feasibilityStudy: [
        {
          name: 'Indication',
          case: 'Indication',
          condition: 'Inclusion',
          caseName: '',
          child: [],
        },
        {
          name: 'Demographics',
          case: 'Demographics',
          condition: 'Inclusion',
          caseName: '',
          child: [
            {
              id: uuid().slice(0, 8),
              condition: 'Gender',
              field: 'gender',
              gender: '',
            },
            {
              id: uuid().slice(0, 8),
              condition: 'Age range',
              field: 'age',
              ageRange: {
                min: 0,
                max: 0,
              },
            },
          ],
        },
      ],
      country: [],
      location: [],
      selectedCountries: [],
      totalConnectedNotConnected: {},
      sites: [],
      sitesContacted: [],
      sitesInitiated: [],
      preferredSites: [],
      patientCount: {
        feasibility: 0,
        location: 0,
      },
      siteCount: {
        feasibility: 0,
        location: 0,
      },
    },
  ],
  cardValue: {
    case: 'Indication',
    condition: '',
    index: 0,
  },
  activeSegment: {
    name: '',
    index: 0,
  },
};

const feasibilityStudySlice = createSlice({
  name: 'feasibilityStudyAction',
  initialState: initialState,
  reducers: {
    setCardDetails: (state, { payload }) => {
      state.cardValue = payload;
    },
    setSegments: (state, { payload }) => {
      state.segments = payload;
    },
    setActiveSegment: (state, { payload }) => {
      state.activeSegment = payload;
    },
    resetSegments: (state) => {
      state.segments = initialState.segments;
      state.cardValue = {
        case: 'Indication',
        condition: '',
        index: 0,
      };
      state.activeSegment = {
        name: '',
        index: 0,
      };
    },
  },
});

export const { setCardDetails, setSegments, setActiveSegment, resetSegments } =
  feasibilityStudySlice.actions;

export default feasibilityStudySlice.reducer;
