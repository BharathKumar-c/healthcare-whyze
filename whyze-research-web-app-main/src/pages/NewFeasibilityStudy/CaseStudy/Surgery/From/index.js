import React from 'react';
import PropTypes from 'prop-types';
import {
  addChildInCaseStudy,
  removeChildInCaseStudy,
  updateCaseStudy,
} from '../../../../../components/shared/AddCondition';
import { conditionOptions, periodOptions } from '../../../../../constant';
import { useDispatch, useSelector } from 'react-redux';
import { Row, Space, Tag } from 'antd';
import { v4 as uuid } from 'uuid';
import { ConditionContainer } from '../../../../../components/shared';
import {
  CustomSelect,
  Image,
  Input,
  Select,
  Text,
} from '../../../../../components/basic';
import { Vector } from '../../../../../assets';
import DragIcon from '../../../../../assets/DragIcon.svg';
import AddIcon from '../../../../../assets/AddIcon.svg';
import CloseIcon from '../../../../../assets/CloseIcon.svg';
import Surgery from '..';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';
import { updateConditionChanges } from '../../../../../components/shared/AddCondition';
import { totalPatientFormatter } from '../../../../../utils/calculations';

export default function From({
  totalpatient,
  parentIndex,
  caseDetails,
  isChild,
  parentId,
  operationValue,
  handleAddCondition,
  handleChangeCaseName,
  getPopupContainer,
}) {
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const dispatch = useDispatch();
  const handleAddNewCondition = (id) => {
    const study = JSON.parse(JSON.stringify(segments));
    addChildInCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      id,
      {
        id: uuid().slice(0, 8),
        condition: '',
        operation: 'And',
        child: [],
      },
    );
    dispatch(setSegments(study));
  };

  const handleSelectRelation = (value, id) => {
    const study = JSON.parse(JSON.stringify(segments));
    const newValue = conditionOptions.find((a) => a.key === value).label;
    updateCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      id,
      'operation',
      newValue,
    );
    dispatch(setSegments(study));
  };

  const handleInputChangeTimeValue = (event) => {
    const study = JSON.parse(JSON.stringify(segments));
    updateConditionChanges(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      caseDetails?.id,
      'from',
      {
        period: caseDetails?.from?.period,
        value: event?.target?.value,
      },
    );
    dispatch(setSegments(study));
  };

  const handleSelectPeriod = (value) => {
    const study = JSON.parse(JSON.stringify(segments));
    updateConditionChanges(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      caseDetails?.id,
      'from',
      {
        value: caseDetails?.from?.value,
        period: value,
      },
    );
    dispatch(setSegments(study));
  };

  const handleRemoveCondition = (id) => {
    const study = JSON.parse(JSON.stringify(segments));
    removeChildInCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      parentId,
      id,
    );
    dispatch(setSegments(study));
  };

  return (
    <ConditionContainer
      className={'condition-container'}
      isActive
      isVertical
      isChild={isChild}
    >
      <Space direction="vertical" className="container">
        <div className="container-alignment">
          <div className="drag-drop-image">
            <Image
              alt="drag-icon"
              src={DragIcon}
              width={10}
              height={16}
              preview={false}
            />
          </div>
          <div className="add-image">
            <Image
              className="image"
              alt="add-icon"
              src={AddIcon}
              width={12}
              height={12}
              preview={false}
              onClick={() => handleAddNewCondition(caseDetails.id)}
            />
          </div>
          <div className="condition-container-before-select" />
          <CustomSelect
            className="condition-container-custom-select"
            options={conditionOptions}
            onSelect={(value) => handleSelectRelation(value, caseDetails.id)}
            defaultValue={operationValue}
            getPopupContainer={getPopupContainer}
          />
          <div className="condition-container-after-select" />
          <Text
            className="condition-container-surgery-sub-text"
            text="Had it"
          />
          <div className="condition-container-surgery-input">
            <Input
              id="main_textbox"
              lable="Enter value"
              name={'value'}
              type="number"
              handleInputChange={handleInputChangeTimeValue}
              value={caseDetails?.from?.value}
              onWheel={(e) => e.target.blur()}
            />
          </div>
          <div
            className="condition-container-surgery-select"
            id="condition-container-surgery-select"
          >
            <Select
              ariaLabel={'from-period'}
              lable="Select type"
              value={caseDetails?.from?.period}
              getPopupContainer={() =>
                document.getElementById('condition-container-surgery-select')
              }
              options={periodOptions}
              handleSelect={handleSelectPeriod}
            />
          </div>
          <Text className="indication-text-label" text="ago" />
          <div className="close-icon">
            <Image
              src={CloseIcon}
              alt="close-icon"
              className="image"
              width={20}
              height={20}
              preview={false}
              onClick={() => handleRemoveCondition(caseDetails.id, parentId)}
            />
          </div>
        </div>
        <Row className="indication-no-patient-tag-row">
          <Tag className="indication-no-patient-tag">
            <Vector />
            {totalPatientFormatter(totalpatient || 0)}
          </Tag>
        </Row>
      </Space>
      {caseDetails?.child?.length > 0 && (
        <>
          <Surgery
            isChild
            parentIndex={parentIndex}
            caseDetails={caseDetails}
            handleAddCondition={handleAddCondition}
            handleChangeCaseName={handleChangeCaseName}
          />
        </>
      )}
    </ConditionContainer>
  );
}

From.propTypes = {
  parentIndex: PropTypes.number,
  isChild: PropTypes.bool,
  totalpatient: PropTypes.number,
  operationValue: PropTypes.string,
  parentId: PropTypes.string,
  getPopupContainer: PropTypes.string,
  caseDetails: PropTypes.object,
  handleAddCondition: PropTypes.func,
  handleChangeCaseName: PropTypes.func,
};

From.defalutProps = {
  parentIndex: 0,
  isChild: false,
  totalpatient: 0,
  operationValue: '',
  parentId: '',
  getPopupContainer: '',
  caseDetails: null,
  handleAddCondition: () => {},
  handleChangeCaseName: () => {},
};
