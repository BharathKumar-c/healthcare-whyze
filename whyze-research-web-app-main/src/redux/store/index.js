import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
