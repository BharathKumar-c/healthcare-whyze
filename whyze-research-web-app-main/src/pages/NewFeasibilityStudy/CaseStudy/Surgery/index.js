import React, { useState } from 'react';
import { Image, Input } from '../../../../components/basic';
import AddIcon from '../../../../assets/AddIcon.svg';
import PropTypes from 'prop-types';
import From from './From';
import AddCondition from '../../../../components/shared/AddCondition';
import { surgeryConditionOptions } from '../../../../constant';
import { surgeryConstants } from '../../../../constant/ConstantTexts';

const Surgery = (props) => {
  const [isError, setIsError] = useState(false);

  const handleError = () => {
    setIsError(true);
  };

  const handleChangeCaseName = (e) => {
    props.handleChangeCaseName(e, props.parentIndex);
    setIsError(false);
  };

  return (
    <div className="indication" id="indication">
      {!props.isChild && (
        <>
          <div className="title-input">
            <Input
              id="main_textbox"
              lable={surgeryConstants.surgeryInputLabel}
              handleInputChange={handleChangeCaseName}
              value={props?.caseDetails?.caseName}
              isError={isError}
              errorText={surgeryConstants.surgeryInputErrorMessage}
            />
          </div>
          <div style={{ height: '30px' }}>
            <Image
              style={{ cursor: 'pointer' }}
              alt="surgery-add-icon"
              preview={false}
              width={12}
              height={12}
              src={AddIcon}
              onClick={() =>
                props.caseDetails.caseName !== ''
                  ? props.handleAddCondition(props.parentIndex)
                  : handleError()
              }
            />
          </div>
        </>
      )}
      {props.caseDetails?.child?.map((c, index) => (
        <div key={`${index}_${c.condition}_${c.operation}`}>
          {props.isChild ? (
            <div
              className={
                c.condition === '' ? 'vertical-line-condition' : 'vertical-line'
              }
            />
          ) : (
            <div
              data-testid="vertical-line"
              style={{
                borderLeft: `2px ${
                  c.condition === '' ? 'dashed' : 'solid'
                } #D3D6D9`,
                height: '32px',
              }}
            />
          )}
          {c.condition === '' ? (
            <AddCondition
              caseDetails={c}
              isChild={props.isChild}
              parentId={props.caseDetails.id}
              parentIndex={props.parentIndex}
              value={c.operation}
              addConditionOptions={surgeryConditionOptions}
              handleAddCondition={props.handleAddCondition}
              handleChangeCaseName={props.handleChangeCaseName}
              getPopupContainer={'indication'}
              conditionName="surgery"
            />
          ) : c.condition === 'from' ? (
            <From
              totalpatient={c?.count}
              parentIndex={props.parentIndex}
              caseDetails={c}
              isChild={props.isChild}
              parentId={props.caseDetails.id}
              operationValue={c.operation}
              handleAddCondition={props.handleAddCondition}
              handleChangeCaseName={props.handleChangeCaseName}
              getPopupContainer={'indication'}
            />
          ) : (
            ''
          )}
        </div>
      ))}
    </div>
  );
};

Surgery.propTypes = {
  caseDetails: PropTypes.object,
  parentIndex: PropTypes.number,
  isChild: PropTypes.bool,
  handleAddCondition: PropTypes.func,
  handleChangeCaseName: PropTypes.func,
};

Surgery.defaultProps = {
  caseDetails: null,
  parentIndex: 0,
  isChild: false,
  handleAddCondition: () => {},
  handleChangeCaseName: () => {},
};

export default Surgery;
