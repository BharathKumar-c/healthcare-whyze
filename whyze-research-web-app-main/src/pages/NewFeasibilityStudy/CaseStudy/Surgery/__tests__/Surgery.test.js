import { act, fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';

import Surgery from '..';
import { surgeryConstants } from '../../../../../constant/ConstantTexts';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';
import store from '../../../../../redux/store';

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}

const caseDetails = {
  name: 'Surgery',
  case: 'Surgery',
  condition: 'Inclusion',
  caseName: '',
  child: [],
};

describe('Test cases for medication', () => {
  test('Create the snapshots for Medication', () => {
    const { container, store } = renderWithContext(<Surgery />);

    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );
      segments[activeSegment.index].feasibilityStudy.push(caseDetails);
      store.dispatch(setSegments(segments));
    });

    expect(container).toMatchSnapshot();
  });

  test('Should render the input field correctly', async () => {
    renderWithContext(<Surgery />);
    const inputField = screen.getByRole('textbox');

    expect(inputField).toBeInTheDocument();
    expect(inputField.value).toMatch('');

    fireEvent.change(inputField, {
      target: { value: 'Surgery' },
    });
    expect(inputField.value).toMatch('Surgery');
  });

  test('Should render the Add icon correctly', () => {
    renderWithContext(<Surgery />);

    const addIcon = screen.getByAltText('surgery-add-icon');
    expect(addIcon).toBeInTheDocument();
  });

  test('Should render error text when click add icon without caseName correctly', () => {
    renderWithContext(<Surgery caseDetails={caseDetails} />);

    const addIcon = screen.getByAltText('surgery-add-icon');
    fireEvent.click(addIcon);

    const errorText = screen.getByText(
      surgeryConstants.surgeryInputErrorMessage,
    );
    expect(errorText).toBeInTheDocument();
  });

  test('Should render when click add icon with caseName correctly', async () => {
    const caseDetailsWithCaseName = {
      name: 'Surgery',
      case: 'Surgery',
      condition: 'Inclusion',
      caseName: 'Surgery',
      child: [],
    };

    const handleAddCondition = jest.fn();
    renderWithContext(
      <Surgery
        caseDetails={caseDetailsWithCaseName}
        parentIndex={2}
        handleAddCondition={handleAddCondition}
      />,
    );

    const addIcon = screen.getByAltText('surgery-add-icon');
    fireEvent.click(addIcon);

    expect(handleAddCondition).toBeCalledTimes(1);
  });

  test('Should render vertical line when surgery has child with condition', () => {
    const caseDetailsWithCaseName = {
      name: 'Surgery',
      case: 'Surgery',
      condition: 'Inclusion',
      caseName: '',
      child: [
        {
          id: '348f4dbe',
          condition: 'from',
          operation: 'And',
          child: [],
          from: {
            period: 'day',
            value: '10',
          },
        },
      ],
    };
    const { container } = renderWithContext(
      <Surgery caseDetails={caseDetailsWithCaseName} isChild />,
    );

    const verticalLine = container.getElementsByClassName('vertical-line');
    expect(verticalLine[0]).toBeInTheDocument();
  });

  test('Should render vertical line when surgery has child without condition', () => {
    const caseDetailsWithCaseName = {
      name: 'Surgery',
      case: 'Surgery',
      condition: 'Inclusion',
      caseName: '',
      child: [
        {
          id: '348f4dbe',
          condition: '',
          operation: 'And',
          child: [],
        },
      ],
    };
    const { container } = renderWithContext(
      <Surgery caseDetails={caseDetailsWithCaseName} isChild />,
    );

    const verticalLine = container.getElementsByClassName(
      'vertical-line-condition',
    );
    const addButton = screen.getByText('+ Add other condition details');
    expect(verticalLine[0]).toBeInTheDocument();
    expect(addButton).toBeInTheDocument();
  });

  test('Should render From component when surgery has from child component', () => {
    const caseDetailsWithCaseName = {
      name: 'Surgery',
      case: 'Surgery',
      condition: 'Inclusion',
      caseName: '',
      child: [
        {
          id: '348f4dbe',
          condition: 'from',
          operation: 'And',
          child: [],
          from: {
            period: 'day',
            value: '10',
          },
        },
      ],
    };
    const { container } = renderWithContext(
      <Surgery caseDetails={caseDetailsWithCaseName} isChild />,
    );

    const fromInputFields = container.querySelector('input[name="value"]');
    expect(fromInputFields).toBeInTheDocument();
  });

  test('Should render vertical dashed line', () => {
    const caseDetailsWithoutCaseName = {
      name: 'Surgery',
      case: 'Surgery',
      condition: 'Inclusion',
      caseName: '',
      child: [
        {
          id: 'cc9e90ff',
          condition: '',
          operation: 'And',
          child: [],
        },
      ],
    };
    renderWithContext(<Surgery caseDetails={caseDetailsWithoutCaseName} />);
    const verticalLine = screen.getByTestId('vertical-line');

    expect(verticalLine).toHaveAttribute(
      'style',
      'border-left: 2px dashed #D3D6D9; height: 32px;',
    );
  });

  test('Should render vertical solid line', () => {
    const caseDetailsWithCaseName = {
      name: 'Surgery',
      case: 'Surgery',
      condition: 'Inclusion',
      caseName: '',
      child: [
        {
          id: '348f4dbe',
          condition: 'from',
          operation: 'And',
          child: [],
          from: {
            period: 'day',
            value: '10',
          },
        },
      ],
    };
    renderWithContext(<Surgery caseDetails={caseDetailsWithCaseName} />);
    const verticalLine = screen.getByTestId('vertical-line');

    expect(verticalLine).toHaveAttribute(
      'style',
      'border-left: 2px solid #D3D6D9; height: 32px;',
    );
  });
});
