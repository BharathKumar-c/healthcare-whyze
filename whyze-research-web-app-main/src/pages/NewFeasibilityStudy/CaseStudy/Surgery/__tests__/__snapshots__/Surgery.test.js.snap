// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for medication Create the snapshots for Medication 1`] = `
<div>
  <div
    class="indication"
  >
    <div
      class="title-input"
    >
      <div
        id="main_textbox"
      >
        <span
          class="ant-typography input-lable"
        >
          Surgery criteria
        </span>
        <input
          class="ant-input"
          data-testid="main_textbox"
          placeholder=""
          style="border-radius: 0px;"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      style="height: 30px;"
    >
      <div
        class="ant-image"
        style="width: 11px; height: 11px;"
      >
        <img
          alt="surgery-add-icon"
          class="ant-image-img"
          height="11"
          src="AddIcon.svg"
          style="height: 11px; cursor: pointer;"
          width="11"
        />
      </div>
    </div>
  </div>
</div>
`;
